src/lib/utils/application-commands/ApplicationCommandRegistry.ts

```diff
@@ -0,0 +1,466 @@
+
+		for (const entry of flattened) {
+			try {
+				BigInt(entry);
+				this.debug(`Registering id "${entry}" to internal chat input map`);
+			} catch {
+				// Don't be silly, save yourself the headaches and do as we say
+				this.debug(`Registering name "${entry}" to internal chat input map`);
+				this.warn(
+					`Registering the chat input command "${entry}" using a name *and* trying to bypass this warning by calling "addChatInputCommandIds" is not recommended.`,
+					'Please use the "addChatInputCommandIds" method with a valid command id instead.'
+				);
+			}
+			this.chatInputCommands.add(entry);
+		}
+
+		return this;
+	}
+
+	public addContextMenuCommandIds(...commandIds: string[] | string[][]) {
+		const flattened = commandIds.flat(Infinity) as string[];
+
+		for (const entry of flattened) {
+			try {
+				BigInt(entry);
+				this.debug(`Registering id "${entry}" to internal context menu map`);
+			} catch {
+				this.debug(`Registering name "${entry}" to internal context menu map`);
+				// Don't be silly, save yourself the headaches and do as we say
+				this.warn(
+					`Registering the context menu command "${entry}" using a name *and* trying to bypass this warning by calling "addContextMenuCommandIds" is not recommended.`,
+					'Please use the "addContextMenuCommandIds" method with a valid command id instead.'
+				);
+			}
+			this.contextMenuCommands.add(entry);
+		}
+
+		return this;
+	}
+
+	protected async runAPICalls(
+		applicationCommands: ApplicationCommandManager,
+		globalCommands: Collection<string, ApplicationCommand>,
+		guildCommands: Map<string, Collection<string, ApplicationCommand>>
+	) {
+		this.debug(`Preparing to process ${this.apiCalls.length} possible command registrations / updates...`);
+
+		const results = await Promise.allSettled(
+			this.apiCalls.map((call) => this.handleAPICall(applicationCommands, globalCommands, guildCommands, call))
+		);
+
+		const errored = results.filter((result) => result.status === 'rejected') as PromiseRejectedResult[];
+
+		if (errored.length) {
+			this.error(`Received ${errored.length} errors while processing command registrations / updates`);
+
+			for (const error of errored) {
+				this.error(error.reason.stack ?? error.reason);
+			}
+		}
+	}
+
+	private async handleAPICall(
+		commandsManager: ApplicationCommandManager,
+		globalCommands: Collection<string, ApplicationCommand>,
+		allGuildsCommands: Map<string, Collection<string, ApplicationCommand>>,
+		apiCall: InternalAPICall
+	) {
+		const { builtData, registerOptions } = apiCall;
+		const commandName = builtData.name;
+		const behaviorIfNotEqual = registerOptions.behaviorWhenNotIdentical ?? getDefaultBehaviorWhenNotIdentical();
+
+		const findCallback = (entry: ApplicationCommand) => {
+			// If the command is a chat input command, we need to check if the entry is a chat input command
+			if (apiCall.type === 'chat_input' && entry.type !== 'CHAT_INPUT') return false;
```

Comment:
Since when are string types... lowercase?

Reply:
apiCall.type is our own type, unrelated to discord.js's.

Reply:
Why is it not an enum?